connect/src/main/java/com/stripe/android/connect/PayoutsView.kt

```diff
@@ -2,49 +2,38 @@ package com.stripe.android.connect
 
 import android.content.Context
 import android.util.AttributeSet
-import android.webkit.WebView
 import android.widget.FrameLayout
 import androidx.annotation.RestrictTo
-import com.stripe.android.connect.webview.StripeConnectWebViewClient
+import com.stripe.android.connect.webview.StripeConnectWebViewContainer
+import com.stripe.android.connect.webview.StripeConnectWebViewContainerImpl
 
 @PrivateBetaConnectSDK
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-class PayoutsView @JvmOverloads constructor(
+class PayoutsView private constructor(
     context: Context,
-    attrs: AttributeSet? = null,
-    defStyleAttr: Int = 0,
-    embeddedComponentManager: EmbeddedComponentManager? = null
-) : FrameLayout(context, attrs, defStyleAttr) {
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    webViewContainerBehavior: StripeConnectWebViewContainerImpl,
+) : FrameLayout(context, attrs, defStyleAttr),
+    StripeConnectWebViewContainer by webViewContainerBehavior {
 
-    private var stripeWebViewClient: StripeConnectWebViewClient? = null
+    @JvmOverloads
+    constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0,
+        embeddedComponentManager: EmbeddedComponentManager? = null,
+    ) : this(
+        context,
+        attrs,
+        defStyleAttr,
+        StripeConnectWebViewContainerImpl(
+            embeddedComponent = StripeEmbeddedComponent.PAYOUTS,
+            embeddedComponentManager = embeddedComponentManager
+        )
+    )
 
     init {
-        inflate(getContext(), R.layout.stripe_connect_webview, this)
-
-        embeddedComponentManager?.let {
-            configureWebView(it)
-        }
-    }
-
-    /**
-     * Set the [EmbeddedComponentManager] to use for this view.
-     * Must be called when this view is created via XML.
-     * Cannot be called more than once per instance.
-     */
-    fun setEmbeddedComponentManager(embeddedComponentManager: EmbeddedComponentManager) {
-        if (stripeWebViewClient != null) {
-            throw IllegalStateException("EmbeddedComponentManager already set")
-        }
-        configureWebView(embeddedComponentManager)
-    }
-
-    private fun configureWebView(embeddedComponentManager: EmbeddedComponentManager) {
-        val webView = findViewById<WebView>(R.id.stripe_web_view)
-        stripeWebViewClient = StripeConnectWebViewClient(
-            embeddedComponentManager,
-            StripeEmbeddedComponent.PAYOUTS,
-        ).apply {
-            configureAndLoadWebView(webView)
-        }
+        webViewContainerBehavior.initializeView(this)
```

Comment:
👀 Unfortunately we can't pass the view into the behavior constructor (because there'd be a cycle), hence the need for this `initializeView()` function